nemo_skills/pipeline/utils/cluster.py

```diff
@@ -198,18 +199,14 @@ def get_env_variables(cluster_config):
                 LOG.info(f"Optional environment variable {env_var_name} not found in user environment; skipping.")
                 _logged_optional_env_vars.add(env_var_name)
 
-    # replace placeholders with actual env var values from the environment where
-    # the job is being launched, not where its being run.
-    for key, value in env_vars.items():
-        if isinstance(value, str) and "$" in value:
-            if key in os.environ:
-                env_vars[key] = os.path.expandvars(value)
-                LOG.info(
-                    f"Resolved environment variable {key} inside the placeholder value: {value} with {env_vars[key]}"
-                )
-            else:
-                raise ValueError(f"Cannot resolve environment variable {key} inside the placeholder value: {value}")
-
+    # Unless NGC_API_KEY is explicitly set we will populate it to be equal to NVIDIA_API_KEY
+    if "NGC_API_KEY" not in env_vars:
+        if "NVIDIA_API_KEY" in env_vars:
+            env_vars["NGC_API_KEY"] = env_vars["NVIDIA_API_KEY"]
+            LOG.info("Populating NGC_API_KEY to be equal to NVIDIA_API_KEY")
+        else:
+            LOG.warning("NGC_API_KEY is not set, but NVIDIA_API_KEY is not set either. "
```

Comment:
I don't think we need a warning here. It's perfectly common situation not to have it set if people are self-hosting models